### DISCUSSION TITLE
Material_ID_texture sampler2D  for post-processing

### DISCUSSION THREAD

**Discussion by goosegarden** [godotengine/godot-proposals#9910] [3D]
Adding a Material_ID sampler2D texture for post-processing. 
It would display objects as flat-coloured with a different color for each material_id, so the color can be used to identify a specific mesh surface in the post-processing stage.
In addition  to depth_texture, normal_texture and velocity_texture, it would give more possibilities for custom layered post-processing, by allowing to apply a specific post-processing effect to a specific mesh, or a specific surface.

**Comment by Calinou** [comment]
As far as I know, this can't be done efficiently in a forward renderer, only in a deferred renderer. If this is the case, then this proposal would depend on https://github.com/godotengine/godot-proposals/issues/8295 being implemented first.

**Reply by goosegarden** [reply]
What i suggest doesn't seem to be related to the proposal you mention. Even though G-Buffers are often used for deferred rendering, the point of this ObjectID buffer is not deferred rendering, it would be used for post-processing effects, more specifically targeted at non-photorealistic rendering.The same way, for example, the depth_texture sampler2D that can  be accessed with hint_depth_texture, or normal_texture sampler2D that can be accessed with hint_normal_roughness_texture, can be used in post-processing (like, for a sobel filter) . I dont know where in  the engine these textures are created, what i suggest is adding another one, to allow to identify a specific object in the post-process stage, so that some custom post process effects can be applied on  specific objects rather than on the whole scene.
